src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -556,29 +561,59 @@ public void SetMinMaxSize(Size minSize, Size maxSize)
 
         public void Resize(Size value, WindowResizeReason reason)
         {
-            if (WindowState != WindowState.Normal)
-                return;
-
             int requestedClientWidth = (int)(value.Width * RenderScaling);
             int requestedClientHeight = (int)(value.Height * RenderScaling);
 
-            GetClientRect(_hwnd, out var clientRect);
+            GetClientRect(_hwnd, out var currentClientRect);
+            if (currentClientRect.Width == requestedClientWidth && currentClientRect.Height == requestedClientHeight)
+            {
+                // Don't update our window position if the client size is already correct. This leads to Windows updating our
+                // "normal position" (i.e. restored bounds) to match our maximised or areo snap size, which is incorrect behaviour.
+                // We only want to proceed with this method if the new size is coming from Avalonia.
+                return;
+            }
 
-            // do comparison after scaling to avoid rounding issues
-            if (requestedClientWidth != clientRect.Width || requestedClientHeight != clientRect.Height)
+            if (_lastWindowState == WindowState.FullScreen)
```

Comment:
We still can't resize a fullscreen window, but I don't think this will ever be supported for the reasons outlined in the comment here. This is not a regression, it's the same behaviour as on master.